src/main/java/ch/njol/skript/classes/data/SkriptClasses.java

```diff
@@ -376,6 +377,23 @@ public String toVariableNameString(final Timeperiod o) {
 						"subtract a day from {_yesterday}",
 						"# now {_yesterday} represents the date 24 hours before now")
 				.since("1.4")
+				.parser(new Parser<Date>() {
+					@Override
+					@Nullable
+					public Date parse(String input, ParseContext context) {
+						return TimeParser.parseDate(input);
```

Comment:
Did you want to restrict this to only `parsed as`? I ask as your example in the description uses `parsed as`.